app/grandchallenge/reader_studies/forms.py

```diff
@@ -741,8 +744,59 @@ def create_ground_truth(self):
         Answer.objects.filter(
             question__reader_study=self._reader_study
         ).update(score=None)
-        bulk_assign_scores_for_reader_study.apply_async(
-            kwargs={"reader_study_pk": self._reader_study.pk}
+
+        on_commit(
+            bulk_assign_scores_for_reader_study.signature(
+                kwargs={"reader_study_pk": self._reader_study.pk}
+            ).apply_async
+        )
+
+
+class AnswersFromGroundTruthForm(SaveFormInitMixin, Form):
+    def __init__(self, *args, reader_study, request_user, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self._reader_study = reader_study
+        self._user = request_user
+
+        self.helper = FormHelper()
+        self.helper.layout = Layout(
+            FormActions(
+                HTML(
+                    format_html(
+                        '<a class="btn btn-secondary" href="{ground_truth_url}">Cancel</a>',
+                        ground_truth_url=reverse(
+                            "reader-studies:ground-truth",
+                            kwargs={"slug": reader_study.slug},
+                        ),
+                    )
+                ),
+                Submit(
+                    "submit",
+                    "Yes, create Answers for Me",
+                    css_class="btn btn-primary",
+                ),
+            )
+        )
+
+    def clean(self):
+        if Answer.objects.filter(
+            question__reader_study=self._reader_study,
+            creator=self._user,
+            is_ground_truth=False,
+        ).exists():
+            raise ValidationError(
+                "User already has answers. Delete these first"
+            )
+
+        return super().clean()
+
+    def schedule_answers_from_ground_truth_task(self):
+        answers_from_ground_truth.apply_async(
```

Comment:
`apply_async` should be always be wrapped in an `on_commit`, otherwise the celery task could run before the changes are committed to the database. Preferred style is how it is below with `on_commit(task.signature(...).apply_async)`.

Reply:
My thought process on not adding it here: there are no changes that need to be commited. Hence it is not wrapped in an on_commit. I could add it, but it would it not be purely decorative?

The only thing I can think of if is that if the transaction fails somehow this would still run. Not sure how it could considering the transaction is empty?

Reply:
That's exactly what this form prevents.

The transaction can fail for a wide array of reasons that are out of your control. We have this guideline as we have faced these sorts of bugs in the past and they are a massive pain. The construction isn't "purely decorative", I would never ask for that and it's a little bit offensive that you consider my suggestions that. I ask for changes as I am the one who has to deal with outages and this construction prevents an entire class of them.

If there is no transaction then the callback will run immediately. If there is a transaction then the callback will be called once everything is done. There are no downsides to adding it.

Reply:
> The transaction can fail for a wide array of reasons that are out of your control. We have this guideline as we have faced these sorts of bugs in the past and they are a massive pain. The construction isn't "purely decorative", I would never ask for that and it's a little bit offensive that you consider my suggestions that. I ask for changes as I am the one who has to deal with outages and this construction prevents an entire class of them.

No offense was ever meant! We sometimes style things because it is done that way, in that particular app, such as adding `View` to some class-based view names and others.

I was honestly wondering if the `on_commit` was useful here, and what rational behind adding it would be. I've added it to the coding guideline.